include/openthread.h

```diff
@@ -154,87 +160,140 @@ extern void otSignalTaskletPending(void);
  */
 const char *otGetVersionString(void);
 
+#ifdef OPENTHREAD_MULTIPLE_INSTANCE
 /**
- * This function initializes the OpenThread library.
+ * This function initializes a new instance of the OpenThread library.
+ *
+ * This function initializes OpenThread and prepares it for subsequent OpenThread API calls.  This function must be
+ * called before any other calls to OpenThread. By default, OpenThread is initialized in the 'enabled' state.
+ *
+ * @param[in]    aInstanceBuffer      The buffer for OpenThread to use for allocating the otInstance structure.
+ * @param[inout] aInstanceBufferSize  On input, the size of aInstanceBuffer. On output, if not enough space for otInstance,
+                                      the number of bytes required for otInstance.
+ *
+ * @retval otInstance*  The new OpenThread instance structure.
+ *
+ */
+otInstance *otInstanceInit(void *aInstanceBuffer, uint64_t *aInstanceBufferSize);
```

Comment:
As an aside, I think we conflate initialization with enable/disable behavior all over the place in the existing API. "Initializing" something is conceptually different from "Enabling" something, especially when that something can then be "Disabled". But that is an issue that can be considered separately from this pull request.